DocumentService/Repositories/DocumentRepository.cs

```diff
@@ -87,21 +107,21 @@ public async Task<bool> SetFileDeleted(Guid id, string userId)
         }
 
         /// <inheritdoc/>
-        public async Task<bool> Update(DocumentInfo documentInfo)
+        public async Task<bool> Update(DocumentDTO documentDTO, Guid id, HttpContext httpContext)
         {
-            var updatedDocumentInfo = await this.GetDocument(documentInfo.DocumentId);
+            var updatedDocumentInfo = await this.GetDocument(id);
             try
             {
                 if (updatedDocumentInfo != null)
                 {
-                    updatedDocumentInfo.UserLastUpdatedById = documentInfo.UserLastUpdatedById;
-                    updatedDocumentInfo.FileName = documentInfo.FileName;
-                    updatedDocumentInfo.FileType = documentInfo.FileType;
-                    updatedDocumentInfo.Description = documentInfo.Description;
-                    updatedDocumentInfo.SubmissionMethod = documentInfo.SubmissionMethod;
-                    updatedDocumentInfo.Language = documentInfo.Language;
-                    updatedDocumentInfo.DocumentTypes = documentInfo.DocumentTypes;
-                    updatedDocumentInfo.DateLastUpdated = DateTime.UtcNow;
+                    updatedDocumentInfo.UserLastUpdatedById = documentDTO.Documents[0].RequesterId;
+                    updatedDocumentInfo.FileName = documentDTO.Documents[0].FileName;
+                    updatedDocumentInfo.FileType = documentDTO.Documents[0].FileType;
+                    updatedDocumentInfo.Description = documentDTO.Documents[0].Description;
+                    updatedDocumentInfo.SubmissionMethod = httpContext.Request.Method;
+                    updatedDocumentInfo.Language = documentDTO.Documents[0].Language;
+                    updatedDocumentInfo.DocumentTypes = documentDTO.Documents[0].DocumentType;
+                    updatedDocumentInfo.DateLastUpdated = GetTimeForNCR();
```

Comment:
There needs to be a chack if documentDTO.Documents is not empty before using it like this.